.github/workflows/publish.yaml

```diff
@@ -4,33 +4,25 @@ on:
   push:
     branches:
       - master
+      - main
+      - track/**
+  pull_request:
+    branches:
+      - master
+      - main
+      - track/**
 
 jobs:
-  charm:
+  publish-charm:
     name: Publish Charm
     runs-on: ubuntu-latest
-
+    # Only publish to charmhub if we are pushing to a special branch or running PRs from something named `branch/*`
+    if: (github.event_name == 'push') ||  (startsWith( github.head_ref, 'branch/' ))
```

Comment:
Can this check be omitted by adding `branch/` to the `push` block above?

Reply:
No I don't think so :( 

The entries in `on` above are applied with `or` (`condition A or condition B or ...`) but we need an `and` (`PR against main/master/track and PR is from branch named branch/*`)

Reply:
Alright, makes sense. Thanks